<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Hypothesis Tests: Non-normal</title>

<script src="site_libs/header-attrs-2.10/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>





<link rel="stylesheet" href="w3.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html"><div class="image-txt-container"><img id="logo" style="width: 400px;" src="figure/umg_logo.png" /></div></a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="index.html">About</a>
</li>
<li>
  <a href="Exercise00.html">R intro</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Exercises
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Exercise01.html">Exe. 1</a>
    </li>
    <li>
      <a href="Exercise02.html">Exe. 2</a>
    </li>
    <li>
      <a href="Exercise03.html">Exe. 3</a>
    </li>
    <li>
      <a href="Exercise04.html">Exe. 4</a>
    </li>
    <li>
      <a href="Exercise05.html">Exe. 5</a>
    </li>
    <li>
      <a href="Exercise06.html">Exe. 6</a>
    </li>
    <li>
      <a href="Exercise07.html">Exe. 7</a>
    </li>
    <li>
      <a href="Exercise08.html">Exe. 8</a>
    </li>
    <li>
      <a href="Exercise09.html">Exe. 9</a>
    </li>
    <li>
      <a href="Exercise10.html">Exe. 10</a>
    </li>
  </ul>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->
<script>
$(document).ready(function() {

  $chunks = $('.fold');

  $chunks.each(function () {

    // add button to source code chunks
    if ( $(this).hasClass('s') ) {
      $('pre.r', this).prepend("<div class=\"showopt\">Show Source</div><br style=\"line-height:22px;\"/>");
      $('pre.r', this).children('code').attr('class', 'folded');
    }

    // add button to output chunks
    if ( $(this).hasClass('o') ) {
      $('pre:not(.r)', this).has('code').prepend("<div class=\"showopt\">Show Output</div><br style=\"line-height:22px;\"/>");
      $('pre:not(.r)', this).children('code:not(r)').addClass('folded');

      // add button to plots
      $(this).find('img').wrap('<pre class=\"plot\"></pre>');
      $('pre.plot', this).prepend("<div class=\"showopt\">Show Plot</div><br style=\"line-height:22px;\"/>");
      $('pre.plot', this).children('img').addClass('folded');

    }
  });

  // hide all chunks when document is loaded
  $('.folded').css('display', 'none')

  // function to toggle the visibility
  $('.showopt').click(function() {
    var label = $(this).html();
    if (label.indexOf("Show") >= 0) {
      $(this).html(label.replace("Show", "Hide"));
    } else {
      $(this).html(label.replace("Hide", "Show"));
    }
    $(this).siblings('code, img').slideToggle('fast', 'swing');
  });
});

</script>


<div id="header">



<h1 class="title toc-ignore">Hypothesis Tests: Non-normal</h1>

</div>


<pre class="r"><code># student
# op_echo=FALSE
# op_eval=FALSE

# teacher
op_echo=TRUE
op_eval=TRUE</code></pre>
<div id="install-an-r-package" class="section level1">
<h1>Install an R package</h1>
<p>R packages are a collection of R functions, complied code and sample data. We install an R package through the function <strong><tt>install.package(package_name)</tt></strong>. Let’s install the package <strong><tt>medicaldata</tt></strong>:</p>
<pre class="r"><code>install.packages(&quot;medicaldata&quot;)</code></pre>
<p>To use this package in our session we need to load it. To accomplish this, we use the <strong><tt>library()</tt></strong> function:</p>
<pre class="r"><code>library(medicaldata)</code></pre>
<pre><code>## Warning: package &#39;medicaldata&#39; was built under R version 4.0.5</code></pre>
<p>from this package we want to retrieve the <strong><tt>polyps</tt></strong> data frame, which contains the systolic blood pressure, the weight and the smoking status for a sample of 500 adults. We use the <strong><tt>library()</tt></strong> to add this data frame to variable in our current environment:</p>
<pre class="r"><code>data(polyps)
str(polyps)
## &#39;data.frame&#39;:    22 obs. of  7 variables:
##  $ participant_id: chr  &quot;001&quot; &quot;002&quot; &quot;003&quot; &quot;004&quot; ...
##  $ sex           : Factor w/ 2 levels &quot;female&quot;,&quot;male&quot;: 1 1 2 1 2 1 1 2 2 2 ...
##  $ age           : num  17 20 16 18 22 13 23 34 50 19 ...
##  $ baseline      : int  7 77 7 5 23 35 11 12 7 318 ...
##  $ treatment     : Factor w/ 2 levels &quot;placebo&quot;,&quot;sulindac&quot;: 2 1 2 1 2 1 2 1 1 1 ...
##  $ number3m      : int  6 67 4 5 16 31 6 20 7 347 ...
##  $ number12m     : num  NA 63 2 28 17 61 1 7 15 44 ...</code></pre>
<p>The database <strong><tt>polyps</tt></strong> contains the results of the randomized controlled trial of sulindac vs. placebo in the prevention of colonic polyps in FAP (Familial Adenomatous Polyposis). You can find more information here <a href="https://htmlpreview.github.io/?https://github.com/higgi13425/medicaldata/blob/master/man/description_docs/polyps_desc.html">polyps</a>.</p>
</div>
<div id="exercises" class="section level1">
<h1>Exercises</h1>
<div id="exercise-5.1" class="section level2">
<h2>Exercise 5.1</h2>
<p>Assume the expression of a particular gene is measures in blood samples of <span class="math inline">\(n=10\)</span> patients with Parkinson’s disease and in <span class="math inline">\(n=10\)</span> healthy controls.</p>
<p>Example data is drawn from exponential distributions as follows:</p>
<pre class="r"><code>set.seed(1223)
n = 10
patients = rexp(n = n, rate = 1/2)
controls = rexp(n = n, rate = 1/1)</code></pre>
Plot the sample values:
<div class="fold s o">
<pre class="r"><code>complete = c(patients, controls)
group = c(rep(1, n), rep(2, n))
plot(group, complete, cex=1.5, xlim=c(0.5, 2.5), axes=FALSE, ylab=&quot;gene expression&quot;)
axis(1, c(1, 2), c(&quot;patients&quot;, &quot;control&quot;))
axis(2)
box()</code></pre>
<p><img src="Exercise05_files/figure-html/unnamed-chunk-5-1.png" width="70%" style="display: block; margin: auto;" /></p>
</div>
Calculate the ranks of sample values. Like above, plot the ranks of sample values.
<div class="fold s o">
<pre class="r"><code>ranked.values = rank(complete)
group = c(rep(1, n), rep(2, n))
plot(group, ranked.values, cex=1.5, xlim=c(0.5, 2.5), axes=FALSE, ylab=&quot;gene expression&quot;)
axis(1, c(1, 2), c(&quot;patients&quot;, &quot;control&quot;))
axis(2)
box()</code></pre>
<p><img src="Exercise05_files/figure-html/unnamed-chunk-6-1.png" width="70%" style="display: block; margin: auto;" /></p>
</div>
<p>Use quantile-quantile plots to check for the assumption of normal distribution.</p>
<div class="fold s o">
<pre class="r"><code>par(mfrow=c(1,2))
qqnorm(patients,main = &#39;patients&#39;)
qqline(patients)
qqnorm(controls,main=&#39;control&#39;,ylab=NA)
qqline(controls)</code></pre>
<p><img src="Exercise05_files/figure-html/unnamed-chunk-7-1.png" width="70%" style="display: block; margin: auto;" /></p>
</div>
<p>Perform Mann-Whitney U test (Wilcoxon Rank Sum test, R function (<strong><tt>wilcox.test</tt></strong>). Conduct a t-test. Compare results from these two tests.</p>
<div class="fold s o">
<pre class="r"><code>
#Null hypothesis is that the distributions of x and y differ by a
#location shift of mu
wilcox.test(patients,controls)
## 
##  Wilcoxon rank sum exact test
## 
## data:  patients and controls
## W = 85, p-value = 0.006841
## alternative hypothesis: true location shift is not equal to 0

#Comparison with t-test result, null hypothesis  equal means
t.test(patients, controls)
## 
##  Welch Two Sample t-test
## 
## data:  patients and controls
## t = 2.9504, df = 13.022, p-value = 0.01125
## alternative hypothesis: true difference in means is not equal to 0
## 95 percent confidence interval:
##  0.4205594 2.7192461
## sample estimates:
## mean of x mean of y 
## 2.3166639 0.7467612</code></pre>
</div>
<!-- we can not trust t-test-->
</div>
<div id="exercise-5.2" class="section level2">
<h2>Exercise 5.2</h2>
<p>Compute the difference between <strong><tt>number12m</tt></strong> level and <strong><tt>baseline</tt></strong> level and add the data to the data frame.</p>
<div class="fold s o">
<pre class="r"><code>polyps$change12m&lt;-polyps$number12m-polyps$baseline</code></pre>
</div>
<p>Check if the differences relative to the <code>treatment=='placebo'</code> are normally distributed and do the same for <code>treatment=='sulindac'</code>.</p>
<div class="fold s o">
<pre class="r"><code>dp&lt;-subset(polyps,treatment==&#39;placebo&#39;)$change12m
ds&lt;-subset(polyps,treatment==&#39;sulindac&#39;)$change12m

#Shapiro-Wilk’s test
shapiro.test(dp)
## 
##  Shapiro-Wilk normality test
## 
## data:  dp
## W = 0.4912, p-value = 1.188e-06
shapiro.test(ds)
## 
##  Shapiro-Wilk normality test
## 
## data:  ds
## W = 0.48965, p-value = 4.83e-06

#QQplot
par(mfrow=c(1,2))
qqnorm(dp,main=&#39;placebo&#39;)
qqline(dp,col=&#39;red&#39;)
qqnorm(ds,main=&#39;sulindac&#39;,ylab=NA)
qqline(ds,col=&#39;red&#39;)</code></pre>
<p><img src="Exercise05_files/figure-html/unnamed-chunk-10-1.png" width="70%" style="display: block; margin: auto;" /></p>
</div>
<p>Calculate the ranks of sample values. Plot the ranks of sample values.</p>
<div class="fold s o">
<pre class="r"><code>ranked_values&lt;-cbind(
  c(rep(1,length(dp)),rep(2,length(ds))),
  rank(c(dp,ds))
)

plot(ranked_values,cex=1.5, xlim=c(0.5, 2.5), axes=FALSE,
     ylab=&quot;number of polyps&quot;,xlab = NA)

axis(1, c(1, 2), c(&quot;patients&quot;, &quot;control&quot;))
axis(2)
box()</code></pre>
<p><img src="Exercise05_files/figure-html/unnamed-chunk-11-1.png" width="70%" style="display: block; margin: auto;" /></p>
</div>
<p>Perform Mann-Whitney U test, in this case you need to set <code>correct=FALSE</code>, because the continuity correction can not be applied as there are ties in our sample. Conduct a t-test. Compare results from these two tests.</p>
<div class="fold s o">
<pre class="r"><code>wilcox.test(ds,dp,correct=FALSE)
t.test(ds,dp)</code></pre>
<pre><code>## 
##  Wilcoxon rank sum test
## 
## data:  ds and dp
## W = 24, p-value = 0.05208
## alternative hypothesis: true location shift is not equal to 0
## 
## 
##  Welch Two Sample t-test
## 
## data:  ds and dp
## t = -0.11864, df = 15.232, p-value = 0.9071
## alternative hypothesis: true difference in means is not equal to 0
## 95 percent confidence interval:
##  -66.3924  59.3823
## sample estimates:
## mean of x mean of y 
## -21.77778 -18.27273</code></pre>
</div>
<p>Interpret the results of the test, keeping in mid the analysis made before.</p>
</div>
<div id="exercise-5.3" class="section level2">
<h2>Exercise 5.3</h2>
<p>Assume there are two different types of disease (type A and B). The mutation rate of a particular gene is <span class="math inline">\(p_1\)</span> under type A and <span class="math inline">\(p_2\)</span> under type B. To compare the mutation rates, tissue samples of <span class="math inline">\(n_A=50\)</span> patients of disease type A and <span class="math inline">\(n_B=100\)</span> patients of disease type B are studied.</p>
<p>Example data is drawn from binomial distributions. Look at the sample values of both patient groups (a value of 0 means no mutation, a value of 1 means there is a mutation).</p>
<pre class="r"><code>n1 = 50
n2 = 100
p1 = 0.1
p2 = 0.15
typeA = rbinom(n = n1, prob = p1, size = 1)
typeB = rbinom(n = n2, prob = p2, size = 1)</code></pre>
<p>Mutation levels are summarized in a 2x2 cross tabulation:</p>
<pre class="r"><code>mutation = c(typeA, typeB)
group = c(rep(&quot;type A&quot;, n1), rep(&quot;type B&quot;, n2))
X = table(mutation, group)
prop.table(X, 2)</code></pre>
<pre><code>##         group
## mutation type A type B
##        0   0.90   0.85
##        1   0.10   0.15</code></pre>
<p>Perform Fisher’s exact test to see whether mutation rates are significantly different between the two groups (<strong><tt>fisher.test</tt></strong>).</p>
<div class="fold s o">
<pre class="r"><code>fisher.test(X)

## Odds ratio
odds_A=p1/(1-p1)
odds_B=p2/(1-p2)
OR=odds_B / odds_A

# Null hypothesis: OR equals to 1
# Null hypothesis : Disease type and mutation status are independent</code></pre>
<pre><code>## 
##  Fisher&#39;s Exact Test for Count Data
## 
## data:  X
## p-value = 0.4558
## alternative hypothesis: true odds ratio is not equal to 1
## 95 percent confidence interval:
##  0.5047588 5.9364737
## sample estimates:
## odds ratio 
##   1.583639</code></pre>
</div>
<p>Change samples sizes <span class="math inline">\(n_1\)</span>, <span class="math inline">\(n_2\)</span> and mutations rates <span class="math inline">\(p_1\)</span>, <span class="math inline">\(p_2\)</span>. Discuss how results change.</p>
<div class="fold s o">
<pre class="r"><code>n1 = 50
n2 = 100
p1 = 0.1
p2 = 0.5
typeA = rbinom(n = n1, prob = p1, size = 1)
typeB = rbinom(n = n2, prob = p2, size = 1)

mutation = c(typeA, typeB)
group = c(rep(&quot;type A&quot;, n1), rep(&quot;type B&quot;, n2))
X = table(mutation, group)

fisher.test(X)
## 
##  Fisher&#39;s Exact Test for Count Data
## 
## data:  X
## p-value = 8.101e-10
## alternative hypothesis: true odds ratio is not equal to 1
## 95 percent confidence interval:
##   5.125226 64.273949
## sample estimates:
## odds ratio 
##   15.59446

odds_A=p1/(1-p1)
odds_B=p2/(1-p2)
odds_B / odds_A
## [1] 9</code></pre>
</div>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "site_libs/mathjax-local/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
